src/reactor.cc

```diff
@@ -0,0 +1,375 @@
+                    helium_storage.quantity());
+}
+
+void Reactor::EnterNotify() {
+  cyclus::Facility::EnterNotify();
+
+  fuel_usage = burn_rate * (fusion_power / MW_to_GW) / seconds_per_year * context()->dt();
+
+  fuel_startup_policy
+      .Init(this, &tritium_storage, std::string("Tritium Storage"),
+            &fuel_tracker, std::string("ss"),
+            startup_inventory,
+            startup_inventory)
+      .Set(fuel_incommod)
+      .Start();
+  blanket_fill_policy
+      .Init(this, &blanket, std::string("Blanket Startup"), &blanket_tracker,
+            std::string("ss"), blanket_size, blanket_size)
+      .Set(blanket_incommod)
+      .Start();
+
+  // Tritium Buy Policy Selection:
+  if (refuel_mode == "schedule") {
+    cyclus::IntDistribution::Ptr active_dist =
+        cyclus::FixedIntDist::Ptr(new cyclus::FixedIntDist(1));
+    cyclus::IntDistribution::Ptr dormant_dist =
+        cyclus::FixedIntDist::Ptr(new cyclus::FixedIntDist(buy_frequency - 1));
+    cyclus::DoubleDistribution::Ptr size_dist =
+        cyclus::FixedDoubleDist::Ptr(new cyclus::FixedDoubleDist(1));
+    fuel_refill_policy
+        .Init(this, &tritium_storage, std::string("Input"), &fuel_tracker,
+              buy_quantity, active_dist, dormant_dist, size_dist)
+        .Set(fuel_incommod);
+  } else if (refuel_mode == "fill") {
+    fuel_refill_policy
+        .Init(this, &tritium_storage, std::string("Input"), &fuel_tracker,
+              std::string("ss"), reserve_inventory, reserve_inventory)
+        .Set(fuel_incommod);
+  } else {
+    throw cyclus::KeyError("Refill mode " + refuel_mode +
+                           " not recognized! Try 'schedule' or 'fill'.");
+    RecordOperationalInfo("Transaction Error",
+                          "Refill mode " + refuel_mode +
+                              " not recognized! Try 'schedule' or 'fill'.");
+  }
+
+  tritium_sell_policy
+      .Init(this, &tritium_excess, std::string("Excess Tritium"))
+      .Set(fuel_incommod)
+      .Start();
+  helium_sell_policy.Init(this, &helium_storage, std::string("Helium-3"))
+      .Set(he3_outcommod)
+      .Start();
+}
+
+std::string Reactor::GetComp(cyclus::Material::Ptr mat) {
+  std::string comp = "{";
+  cyclus::CompMap c = mat->comp()->atom();
+  cyclus::compmath::Normalize(&c, 1);
+  for (std::map<const int, double>::const_iterator it = c.begin();
+       it != c.end(); ++it) {
+    comp = comp + std::string("{") + std::to_string(it->first) +
+           std::string(",") + std::to_string(it->second) + std::string("},");
+  }
+  comp.pop_back();
+  comp = comp + std::string("}");
+  return comp;
+}
+
+void Reactor::SequesterTritium(){
+  if (!tritium_sequestered.empty()){
+    cyclus::Material::Ptr sequestered_mat = tritium_sequestered.Pop();
+    cyclus::CompMap c = sequestered_mat->comp()->atom();
+    cyclus::compmath::Normalize(&c, sequestered_mat->quantity());
```

Comment:
See `MatQuery` in the toolkit for what you want here & in many other places

Reply:
Fixed. Had to leave things the old way (compmap c --> normalize(c, 1)) in startup because matquary doesn't quite do what I need it to there, but everywhere else it should be gone (I'll be removing GetComp() in a minute)